tracing/src/main/java/com/palantir/tracing/Span.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.tracing;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+import org.immutables.value.Value;
+
+/**
+ * A value class representing a completed Span.
+ */
+@JsonDeserialize(as = ImmutableSpan.class)
+@JsonSerialize(as = ImmutableSpan.class)
+@Value.Immutable
+@Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE)
+public abstract class Span {
```

Comment:
I'm a little confused: why do we need both TraceState and Span if they contain mostly the same things?